src/main/java/org/graylog/integrations/notifications/types/SlackEventNotificationConfig.java

```diff
@@ -41,140 +42,165 @@
-
-		@JsonProperty(FIELD_CHANNEL)
-		public abstract SlackEventNotificationConfig.Builder channel(String channel);
-
-		@JsonProperty(FIELD_CUSTOM_MESSAGE)
-		public abstract SlackEventNotificationConfig.Builder customMessage(String customMessage);
-
-		@JsonProperty(FIELD_USER_NAME)
-		public abstract SlackEventNotificationConfig.Builder userName(String userName);
-
-		@JsonProperty(FIELD_NOTIFY_CHANNEL)
-		public abstract SlackEventNotificationConfig.Builder notifyChannel(boolean notifyChannel);
-
-		@JsonProperty(FIELD_LINK_NAMES)
-		public abstract SlackEventNotificationConfig.Builder linkNames(boolean linkNames);
+    @JsonProperty(FIELD_WEBHOOK_URL)
+    @NotBlank
+    public abstract String webhookUrl();
 
-		@JsonProperty(FIELD_ICON_URL)
-		public abstract SlackEventNotificationConfig.Builder iconUrl(String iconUrl);
+    @JsonProperty(FIELD_CHANNEL)
+    @NotBlank
+    public abstract String channel();
 
-		@JsonProperty(FIELD_ICON_EMOJI)
-		public abstract SlackEventNotificationConfig.Builder iconEmoji(String iconEmoji);
+    @JsonProperty(FIELD_CUSTOM_MESSAGE)
+    public abstract String customMessage();
 
+    @JsonProperty(FIELD_USER_NAME)
+    @Nullable
+    public abstract String userName();
 
-		public abstract SlackEventNotificationConfig build();
-	}
+    @JsonProperty(FIELD_NOTIFY_CHANNEL)
+    public abstract boolean notifyChannel();
 
-	@Override
-	public EventNotificationConfigEntity toContentPackEntity(EntityDescriptorIds entityDescriptorIds) {
-		return SlackEventNotificationConfigEntity.builder()
-				.color(ValueReference.of(color()))
-				.webhookUrl(ValueReference.of(webhookUrl()))
-				.channel(ValueReference.of(channel()))
-				.customMessage(ValueReference.of(customMessage()))
-				.userName(ValueReference.of(userName()))
-				.notifyChannel(ValueReference.of(notifyChannel()))
-				.linkNames(ValueReference.of(linkNames()))
-				.iconUrl(ValueReference.of(iconUrl()))
-				.iconEmoji(ValueReference.of(iconEmoji()))
-				.build();
-	}
+    @JsonProperty(FIELD_LINK_NAMES)
+    public abstract boolean linkNames();
+
+    @JsonProperty(FIELD_ICON_URL)
+    @Nullable
+    public abstract String iconUrl();
+
+    @JsonProperty(FIELD_ICON_EMOJI)
+    @Nullable
+    public abstract String iconEmoji();
+
+
+    @Override
+    @JsonIgnore
+    public JobTriggerData toJobTriggerData(EventDto dto) {
+        return EventNotificationExecutionJob.Data.builder().eventDto(dto).build();
+    }
+
+    public static SlackEventNotificationConfig.Builder builder() {
+        return SlackEventNotificationConfig.Builder.create();
+    }
+
+    @Override
+    @JsonIgnore
+    public ValidationResult validate() {
+        ValidationResult validation = new ValidationResult();
+        final Matcher matcher = pattern.matcher(webhookUrl());
+
+        if (backlogSize() < 0 || backlogSize() > MAX_BACKLOG_MESSAGES) {
+            validation.addError(FIELD_BACKLOG_SIZE, INVALID_BACKLOG_ERROR_MESSAGE);
+        }
```

Comment:
When `backlogSize` is set to zero, we do not limit the backlog at all.  Thus, having an upper limit on `backlogSize` makes no sense.  If the user has decided to set the backlog for their Alert to 300 and we will allow them to get all 300, why would we not allow them to get 100 or 200?

Reply:
the slack client throws Http_error_code 500 , when the message goes beyond a certain limit. I would like to impose a limit, on the number of messages being sent to the slack client, just so we don't break the Slack client. Its a good practice to set the upper limit. The number 50 is empirical and is subject to change as we find new facts.

Reply:
My biggest point here is that it makes no sense to put a max on the backlog limit because the backlog limit can be disabled and the entire backlog will flow through.

The amount of data that gets POSTed to the Slack webhook is entirely dependent on the message template being used.  If the user is dumping the full contents of each backlog message, then 5 messages may be enough to produce an oversized Slack message.  If the user is only including message IDs, then 500 messages might be fine.  But if the user sets their message backlog to N and they can cause all N backlog messages to be sent by disabling the limit, why should we not allow them to set their limit to values from 51 to (N - 1)?

A quick Google search pointed to the [Slack documentation](https://api.slack.com/changelog/2018-04-truncating-really-long-messages), which indicates that any message with more than 40000 characters of text will be truncated on the server side.  I have validated this locally with a 60000-character message template (you can see some of the truncated messages in our test Slack channel).  I should note that these auto-truncated messages returned HTTP 200 when POSTed to the webhook.

Given that Slack will auto-truncate overly large messages, I don't think there's a way for us to come up with a correct upper limit.  It would make more sense (knowing about Slack's auto-truncate behavior) for us to shorten the message on our side before sending it to Slack and put a warning in the logs than to put an arbitrary cap on

Reply:
I will remove the max_backlog_messages as per your advice.